lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -479,6 +534,43 @@ LinearLayout sharedToLinearLayoutLeadingOffset(ArrayRef<int64_t> shape,
   return combineCtaCgaWithShape(tileLayout, shared.getCTALayout(), shape);
 }
 
+// This function convert blockedEncodingAttr to linear layout in a special way.
+// It accompanies the AMDGPUInThreadTranspose pass to transpose K-major tensor
+// into K-minor prior to writing into LDS (shared memory). This conversion
+// treats the sizePerThread as a 2D matrix and has different access pattern.
+//
+// For example, consider the following blocked layout generated by
+// AMDGPUInThreadTranspose: #blocked1 = #triton_gpu.blocked<{sizePerThread =
+// [4, 8], threadsPerWarp = [2, 32], warpsPerCTA = [8, 1], order = [1, 0]}>.
+// Here since sizePerThread is 2D, there could be two ways to traverse it: along
+// the dim of 8 or the dim of 4. The regular toLinearLayout() would go through
+// it from the leading order, i.e. dim of 8, but since we want to transpose it
+// in-thread, we'd want to iterate of the 2nd order, i.e. dim of 4, so that we
+// can pack the element of 4 into a single vector, and AMD backend LLVM compiler
+// will pack elements into consecutive VGPR and therefore achieve high
+// vectorization LDS write and also keep data in K-minor inside LDS.
```

Comment:
We should not say "high vectorization LDS write" because the old way, i.e. traversing the dim of 8, has higher vectorsize of ds_write.
So the ds_write can have short or long vector size, but we are guaranteed to have vectorized ds_read.

Reply:
How about change this 
"and AMD backend LLVM compiler will pack elements into consecutive VGPR and therefore achieve high vectorization LDS write and also keep data in K-minor inside LDS." 
to
"and AMD backend LLVM compiler will pack elements into consecutive VGPR to write data contiguous in K dimension into LDS. In this way we guarantee vectorized ds_read, and ds_write can be vectorized to 64bit or 32bit depending on the block size and number of warps"

Reply:
Sounds good